integration/tests/pricing-table.test.ts

```diff
@@ -48,26 +53,54 @@ testAgainstRunningApps({ withEnv: [appConfigs.envs.withBilling] })('pricing tabl
     await u.po.signIn.goTo();
     await u.po.signIn.signInWithEmailAndInstantPassword({ email: fakeUser.email, password: fakeUser.password });
     await u.po.page.goToRelative('/pricing-table');
-    // We have two plans, so subscribe to the first one
-    await u.po.page.getByText('Get started').first().click();
 
-    // Stripe uses multiple iframes, so we need to find the correct one
-    const frame = u.po.page.frameLocator('iframe[src*="elements-inner-payment"]');
-    await frame.getByLabel('Card number').fill('4242424242424242');
-    await frame.getByLabel('Expiration date').fill('1234');
-    await frame.getByLabel('Security code').fill('123');
-    await frame.getByLabel('ZIP code').fill('12345');
+    await u.po.pricingTable.startCheckout({ planSlug: 'plus' });
+    await u.po.checkout.waitForMounted();
+    await u.po.checkout.fillTestCard();
+    await u.po.checkout.continueWithNewCard();
+    await expect(u.po.page.getByText('Payment was successful!')).toBeVisible();
+  });
 
-    await u.po.page.getByRole('button', { name: 'Pay $' }).click();
+  test('can upgrade to a new plan with saved card', async ({ page, context }) => {
+    const u = createTestUtils({ app, page, context });
+    await u.po.signIn.goTo();
+    await u.po.signIn.signInWithEmailAndInstantPassword({ email: fakeUser.email, password: fakeUser.password });
+    await u.po.page.goToRelative('/pricing-table');
+
+    await u.po.pricingTable.startCheckout({ planSlug: 'pro', shouldSwitch: true });
```

Comment:
Not a super strong opinion, but shouldSwitch is not super obvious. `switchPlans` maybe? Again not super strong opinion here.